ch_pipeline/core/dataquery.py

```diff
@@ -655,6 +658,111 @@ def next(self, ts):
         return inputs
 
 
+class QueryFrequencyMap(task.MPILoggedTask):
+    """Get the CHIME frequency map that was active when this data was collected.
+
+    Attributes
+    ----------
+    cache : bool
+        Only query for the inputs for the first container received. For all
+        subsequent files just return the initial set of inputs. This can help
+        minimise the number of potentially fragile database operations.
+    """
+
+    cache = config.Property(proptype=bool, default=False)
+
+    _cached_inputs = None
+
+    def next(self, ts) -> FrequencyMapSingle:
+        """Generate an input description from the timestream passed in.
+
+        Parameters
+        ----------
+        ts : andata.CorrData
+            Timestream container.
+
+        Returns
+        -------
+        freq_map : FrequencySlotMap
+            Frequency slot map container
+        """
+
+        # Fetch from the cache if we can
+        if self.cache and self._cached_inputs:
+            self.log.debug("Using cached inputs.")
+            return self._cached_inputs
+
+        stream_id = None
+        stream = None
+
+        if mpiutil.rank0:
+            layout.connect_database()
+
+            # Check to see if this timestream has a dataset id, and assign it
+            # if it does. If there is no dataset_id, we will just assume a
+            # default mapping
+            if (did_flag := ts.flags.get("dataset_id", None)) is not None:
```

Comment:
```suggestion
            if (dsid_flag := ts.flags.get("dataset_id", None)) is not None:
```
We usually denote these as dsid/ds_id or similar. did is a little confusing when reading.